contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -12,30 +12,94 @@
+        # TODO: Implement retry logic
+        self.storage = None
 
     def export_metrics(self, metrics):
-        metrics = list(metrics)
-        for metric in metrics:
-            # TODO: implement export to Azure Monitor functionality
-            print(repr(metric))
+        if metrics:
+            for metric in metrics:
+                # No support for histogram aggregations
+                type_ = metric.descriptor.type
+                if type_ != MetricDescriptorType.CUMULATIVE_DISTRIBUTION:
+                    md = metric.descriptor
+                    # Each time series will be uniquely
+                    # identified by it's label values
+                    for time_series in metric.time_series:
+                        # Using stats, time_series should
+                        # only have one point which contains
+                        # the aggregated value
+                        data_point = self.create_data_points(
+                                     time_series, md)[0]
+                        # The timestamp is when the metric was recorded
+                        time_stamp = time_series.points[0].timestamp
+                        # Get the properties using label keys from metric
+                        # and label values of the time series
+                        properties = self.create_properties(time_series, md)
+                        envelope = self.create_envelope(
+                                   data_point, time_stamp, properties)
+                        self._transmit(envelope)
+
+    def create_data_points(self, time_series, metric_descriptor):
+        """Convert an metric's OC time series to list of Azure data points."""
+        data_points = []
+        for point in time_series.points:
+            data_point = DataPoint(ns=metric_descriptor.name,
+                                   name=metric_descriptor.name,
+                                   value=point.value.value)
+            data_points.append(data_point)
+        return data_points
+
+    def create_properties(self, time_series, metric_descriptor):
+        properties = {}
+        # We construct a properties map from the
+        # label keys and values
+        # We assume the ordering is already correct
+        for i in range(len(metric_descriptor.label_keys)):
+            if time_series.label_values[i].value is None:
+                value = "None"
+            else:
+                value = time_series.label_values[i].value
+            properties[metric_descriptor.label_keys[i].key] = value
+        return properties
+
+    def create_envelope(self, data_point, time_stamp, properties):
+        envelope = Envelope(
+            iKey=self.options.instrumentation_key,
+            tags=dict(utils.azure_monitor_context),
+            time=time_stamp.isoformat(),
+        )
+        envelope.name = "Microsoft.ApplicationInsights.Metric"
+        # The ingestion service for Azure Monitor only takes a single
```

Comment:
I don't think the ingestion service only supports single data point per request.

Reply:
You are correct. Fixed to use batch in form of passing in list of envelopes.